aurora/src/test/java/com/market/banica/aurora/AuroraComponentIT.java

```diff
@@ -0,0 +1,853 @@
+    }
+
+    @Test
+    void subscribeToMarket_Should_ForwardToReceiverResponse() throws IOException, InterruptedException {
+        //Arrange
+        publishers.addPublisher("market");
+        createFakeMarketServer();
+
+        createFakeSender();
+
+
+        grpcCleanup.register(senderChannel);
+        grpcCleanup.register(receiverChannel);
+
+        receiverChannel.getState(true);
+        channelManager.addChannel("market", receiverChannel);
+
+        Aurora.AuroraRequest request = Aurora.AuroraRequest.newBuilder().setTopic("market/eggs").build();
+
+        ArrayList<TickResponse> expectedResponses = new ArrayList<>();
+        for (int i = 0; i < 5; i++) {
+            TickResponse response = TickResponse.newBuilder().setGoodName("eggs").build();
+
+            expectedResponses.add(response);
+        }
+
+        ArrayList<Aurora.AuroraResponse> receivedResponses = new ArrayList<>();
+        CountDownLatch latch = new CountDownLatch(1);
+
+        //Act
+        AuroraServiceGrpc
+                .newStub(senderChannel)
+                .subscribe(request, new StreamObserver<Aurora.AuroraResponse>() {
+                    @Override
+                    public void onNext(Aurora.AuroraResponse response) {
+                        receivedResponses.add(response);
+                    }
+
+                    @Override
+                    public void onError(Throwable throwable) {
+                        latch.countDown();
+                    }
+
+                    @Override
+                    public void onCompleted() {
+                        latch.countDown();
+                    }
+                });
+
+        //Assert
+        latch.await();
+        List<TickResponse> unpackedResponses = receivedResponses.stream()
+                .map(response -> {
+                    try {
+                        return response.getMessage().unpack(TickResponse.class);
+                    } catch (InvalidProtocolBufferException e) {
+                        e.printStackTrace();
+                    }
+                    return null;
+                })
+                .collect(Collectors.toList());
+        assertEquals(expectedResponses, unpackedResponses);
+        channelManager.removeChannel("market");
+        publishers.deletePublisher("market");
+    }
+
+    @Test
+    void subscribeToOrderbook_Should_ForwardToReceiverErrorDueNoMapping() throws IOException, InterruptedException {
+        //Arrange
+        publishers.addPublisher("orderbook");
+        createFakeOrderBookServer();
+        createFakeSender();
+
+        grpcCleanup.register(senderChannel);
+        grpcCleanup.register(receiverChannel);
+
+        receiverChannel.getState(true);
+        channelManager.addChannel("orderbook", receiverChannel);
+
+        Aurora.AuroraRequest request = Aurora.AuroraRequest.newBuilder().setTopic("orderbook/something").build();
+
+        final boolean[] inError = {false};
```

Comment:
why is this an array ?

Reply:
Because variable is accessed from within inner class needs to be final or effectively final .

I can make an final singletonList as an alternative.